packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -105,13 +105,33 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
 
     let flatData: { [key: string]: string } = {};
     let data: { [key: string]: string } = {};
-    const call: AwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
+    const call: EncodedAwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
 
     if (call) {
-      const awsService = new (AWS as any)[call.service]({
-        apiVersion: call.apiVersion,
-        region: call.region,
-      });
+
+      let awsService: any;
+      if(call.assumedRole){
+        const sts = new AWS.STS();
+        const { Credentials } = await sts
```

Comment:
Wondering if a better approach here is to use [`ChainableTemporaryCredentials`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/ChainableTemporaryCredentials.html) instead?

Reply:
Oh I didn't actually know about this! This looks really nice 👍  

But now I am stuck on how to test this 😢 . mocking `assumeRole` (although it is used internally) no longer works

Reply:
It seems to be a general problem: 

https://github.com/aws/aws-cdk/blob/1ed119e2cdbc37666616f6666b0edb12c2c9ea89/packages/aws-cdk/test/api/sdk-provider.test.ts#L233-L234

but I don't really see how I could use that here

Reply:
@eladb Do you see any way to test this using `ChainableTemporaryCredentials`?

Reply:
> @eladb Do you see any way to test this using `ChainableTemporaryCredentials`?

@eladb